### DISCUSSION TITLE
Remote DevContainer Git HTTPS push/pull error - remote: Repository not found.

### DISCUSSION THREAD

**Discussion by MVPavan** [devcontainers/community#131] [Q&A] [answered]
I use Dev Container on my remote machine with VSCode and corporate GIT with HTTPS.

Authenticated using VSCode + Github(web) including private repo authorizations.

Working:

1. On my Windows laptop, using VSCode IDE I can stage, commit, and push/pull to all my corporate private repos,  with no issues.
2. On my remote desktop (Linux), using VSCode IDE I can stage, commit, and push/pull to all my corporate private repos, with no issues.
3. On my remote desktop (Linux) in dev containers, using bash shell I can stage, commit, and push/pull to all my corporate private repos,  with no issues.

Not Working:

1. On my remote in dev containers, I can stage and commit but push/pull is failing with an error `remote: Repository not found.`

Note: I have to use HTTPS, not SSH.

I appreciate any help. Thanks.

**Comment by samruddhikhandale** [comment, marked as answer]
Hi 👋 

The issue you're experiencing might be related to the authentication in your dev container environment. When you're working inside a dev container, the container might not have access to the same authentication tokens or SSH keys that your host machine does.

Here are a few things you can try:

- Check your git configuration in the dev container: Run git config --list inside the dev container to see what user and email are set. Make sure they match with your GitHub account.
- Re-authenticate inside the dev container: You might need to authenticate again inside the dev container. You can do this by running gh auth login and following the prompts.
- Use credential helper: Git has a credential helper that can store your credentials, so you don't have to enter them every time. You can enable it by running git config --global credential.helper store inside the dev container.
- Check your remote URL: Make sure the remote URL is correct. You can check it by running git remote -v inside the dev container. The URL should be in the format https://github.com/username/repo.git.
- Check access to the private repo: Make sure the GitHub account you're using inside the dev container has access to the private repo. You can check this by going to the repo on GitHub and checking the "Settings > Manage access" page.

Let us know if any of the solutions mentioned help, thanks!

**Reply by MVPavan** [reply]
Hi,

gh auth login worked.

It's weird though, that the pull requests extension worked without any extra authentication, but has an issue with pull/push.

Anyway, it works now, thanks!
